run_node.sh

```diff
@@ -2,8 +2,390 @@
+                --mount type=bind,source="${HOST_BASE_PATH}",target=${BASE_PATH} \
+                --name "${NAME}" \
+                --restart unless-stopped \
+                -d "${ALEPH_IMAGE}"
+}
+
+run_archivist () {
+    if [ -n "${MAINNET}" ] && [ "${MAINNET}" = 'true' ]
+    then
+        source env/archivist_mainnet
+        eval "echo \"$(cat env/archivist_mainnet)\"" > env/archivist_mainnet.env
+        ENV_FILE="env/archivist_mainnet.env"
+    else
+        source env/archivist
+        eval "echo \"$(cat env/archivist)\"" > env/archivist.env
+        ENV_FILE="env/archivist.env"
+    fi
+
+
+    PROXY_PORT=${PROXY_PORT:-$PORT}
+
+    if [[ -n "${PUBLIC_DNS}" ]]
+    then
+        PUBLIC_ADDR="/dns4/${PUBLIC_DNS}/tcp/${PROXY_PORT}"
+    else
+        PUBLIC_ADDR="/ip4/${PUBLIC_IP}/tcp/${PROXY_PORT}"
+    fi
+
+    echo "Running with public P2P address: ${PUBLIC_ADDR}"
+
+    PORT_MAP="${PORT}:${PORT}"
+
+    docker run --env-file ${ENV_FILE} \
+                --env PUBLIC_ADDR="${PUBLIC_ADDR}" \
+                -p ${RPC_PORT_MAP} \
+                -p ${PORT_MAP} \
+                -p ${METRICS_PORT_MAP} \
+                -u "$(id -u):$(id -g)" \
+                --mount type=bind,source=${HOST_BASE_PATH},target=${BASE_PATH} \
+                --name ${NAME} \
+                --restart unless-stopped \
+                -d "${ALEPH_IMAGE}"
+}
+
+
+# The defaults
+NAME="aleph-node-$(xxd -l "16" -p /dev/urandom | tr -d " \n" ; echo)"
+BASE_PATH="/data"
+HOST_BASE_PATH="${HOME}/.alephzero"
+DB_SNAPSHOT_FILE="db_backup.tar.gz"
+DB_SNAPSHOT_URL="http://db.test.azero.dev.s3-website.eu-central-1.amazonaws.com/latest.html"
+MAINNET_DB_SNAPSHOT_URL="http://db.azero.dev.s3-website.eu-central-1.amazonaws.com/latest.html"
+DB_SNAPSHOT_PATH="chains/testnet/"     # testnet by default
+CHAINSPEC_FILE="testnet_chainspec.json"
+
+while [[ $# -gt 0 ]]; do
+    case "$1" in
+        -h | --help) # display Help
+            Help
+            exit;;
+        --archivist) # Run as an archivist
+            ARCHIVIST=true
+            shift;;
+        --ip)
+            PUBLIC_IP="$2"
+            shift 2;;
+        --dns)
+            PUBLIC_DNS="$2"
+            shift 2;;
+        -n | --name) # Enter a name
+            NAME="$2"
+            shift 2;;
+        -d | --data_dir) # Choose the data directory
+            HOST_BASE_PATH="$2"
+            shift 2;;
+        --mainnet) # Join the mainnet
+            DB_SNAPSHOT_PATH="chains/mainnet/"
+            CHAINSPEC_FILE="mainnet_chainspec.json"
+            DB_SNAPSHOT_URL="${MAINNET_DB_SNAPSHOT_URL}"
+            CLIAIN_ENDPOINT='wss://ws.azero.dev:443'
+            MAINNET=true
+            shift;;
+        --local) # Bootstrap a local network
+            CHAINSPEC_FILE="local_chainspec.json"
+            CLIAIN_ENDPOINT='wss://ws.azero.dev:443'
+            MAINNET=false
+            LOCAL=true
+            shift;;
+        -i | --image) # Enter a base path
```

Comment:
Do we want to give the possibility to override aleph-node version? I can see it was previously there as well, yet what's the use case we support here. E.g. someone can pass some arbitrary sha here, but then script will download chainspec and snapshot from testnet version anyway. I think this script should support official versions (ie Testnet and Mainnet) only, wdyt?

Reply:
Removed (I might enable it later on, but only for the local deployment -- we'll see)